src/accessors/IIterator.ts

```diff
@@ -1,6 +1,6 @@
-import { IIteratorResult } from './IIteratorResult';
-
 export interface IIterator<T> {
-    next(skip: number): IIteratorResult<T>;
+    next(packageSize?: number): Array<T>;
```

Comment:
@graywolf336 ahh, the iterator protocol. Nice, didn't know that. Then, the object return value makes sense, particularly as the `for ... in...` would actually make it transparent.
The iterator protocol answers also where to add the pagesize:
> A **zero arguments function** that returns an object with two properties
👍